x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java

```diff
@@ -225,7 +237,36 @@ public void onFailure(Exception e) {
 
                 @Override
                 protected void doRun() throws Exception {
-                    handler.messageReceived(request, channel, task);
+                    handler.messageReceived(request, new TransportChannel() {
+
+                        @Override
+                        public String getProfileName() {
+                            return channel.getProfileName();
+                        }
+
+                        @Override
+                        public String getChannelType() {
+                            return channel.getChannelType();
+                        }
+
+                        @Override
+                        public Version getVersion() {
+                            return channel.getVersion();
+                        }
+
+                        @Override
+                        public void sendResponse(TransportResponse response) throws IOException {
+                            String requestId = AuditUtil.extractRequestId(threadContext);
+                            Authentication authentication = securityContext.getAuthentication();
+                            auditTrailService.get().actionResponse(requestId, authentication, action, request, response);
+                            channel.sendResponse(response);
+                        }
+
+                        @Override
+                        public void sendResponse(Exception exception) throws IOException {
+                            channel.sendResponse(exception);
+                        }
+                    }, task);
```

Comment:
This intercepts the response for actions executed on other cluster nodes (when an action on the coordinating node submits requests to other nodes - or even the same node)